nebula_ros/src/continental/continental_ars548_hw_interface_ros_wrapper.cpp

```diff
@@ -0,0 +1,510 @@
+}
+
+void ContinentalARS548HwInterfaceRosWrapper::SetNetworkConfigurationRequestCallback(
+  const std::shared_ptr<continental_srvs::srv::ContinentalArs548SetNetworkConfiguration::Request>
+    request,
+  const std::shared_ptr<continental_srvs::srv::ContinentalArs548SetNetworkConfiguration::Response>
+    response)
+{
+  std::scoped_lock lock(mtx_config_);
+  auto result = hw_interface_.SetSensorIPAddress(request->sensor_ip.data);
+  response->success = result == Status::OK;
+  response->message = (std::stringstream() << result).str();
+}
+
+void ContinentalARS548HwInterfaceRosWrapper::SetSensorMountingRequestCallback(
+  const std::shared_ptr<continental_srvs::srv::ContinentalArs548SetSensorMounting::Request> request,
+  const std::shared_ptr<continental_srvs::srv::ContinentalArs548SetSensorMounting::Response>
+    response)
+{
+  std::scoped_lock lock(mtx_config_);
+
+  auto tf_buffer = std::make_unique<tf2_ros::Buffer>(this->get_clock());
+  auto tf_listener = std::make_unique<tf2_ros::TransformListener>(*tf_buffer);
+
+  geometry_msgs::msg::TransformStamped base_to_sensor_tf;
+  try {
+    base_to_sensor_tf = tf_buffer->lookupTransform(
+      sensor_configuration_.base_frame, sensor_configuration_.frame_id, rclcpp::Time(0),
+      rclcpp::Duration::from_seconds(0.5));
+  } catch (tf2::TransformException & ex) {
+    RCLCPP_ERROR(
+      this->get_logger(), "Could not obtain the transform from the base frame to %s (%s)",
+      sensor_configuration_.frame_id.c_str(), ex.what());
+    response->success = false;
+    response->message = ex.what();
+    return;
+  }
+
+  const auto & quat = base_to_sensor_tf.transform.rotation;
+  geometry_msgs::msg::Vector3 rpy;
+  tf2::Matrix3x3(tf2::Quaternion(quat.x, quat.y, quat.z, quat.w)).getRPY(rpy.x, rpy.y, rpy.z);
+
+  auto result = hw_interface_.SetSensorMounting(
+    base_to_sensor_tf.transform.translation.x -
+      sensor_configuration_.configuration_vehicle_wheelbase,
+    base_to_sensor_tf.transform.translation.y, base_to_sensor_tf.transform.translation.z, rpy.z,
+    rpy.y, request->plug_orientation);
+
+  response->success = result == Status::OK;
+  response->message = (std::stringstream() << result).str();
+}
+
+void ContinentalARS548HwInterfaceRosWrapper::SetVehicleParametersRequestCallback(
+  [[maybe_unused]] const std::shared_ptr<
+    continental_srvs::srv::ContinentalArs548SetVehicleParameters::Request>
+    request,
+  const std::shared_ptr<continental_srvs::srv::ContinentalArs548SetVehicleParameters::Response>
+    response)
+{
+  std::scoped_lock lock(mtx_config_);
+  auto result = hw_interface_.SetVehicleParameters(
+    sensor_configuration_.configuration_vehicle_length,
+    sensor_configuration_.configuration_vehicle_width,
+    sensor_configuration_.configuration_vehicle_height,
+    sensor_configuration_.configuration_vehicle_wheelbase);
+
+  response->success = result == Status::OK;
+  response->message = (std::stringstream() << result).str();
+}
```

Comment:
It might be unexpected behavior for users to have to send an empty service request to send params to the sensor that were defined in the launch file.

It would be nicer if
* params were sent in the service request, or
* the service is removed and the params in the launch file are set on startup without a service call

Reply:
In [423112d](https://github.com/tier4/nebula/pull/123/commits/423112d69d8bfec2446adae90cf4ce20dae5c505)
I added an option to set the parameters via the services. Basically:

vehicle parameters: if no parameters are given to the service (`ros2 service call name type {}`), it will use the parameters from the launcher. If parameters are given, it will use those (`ros2 service call name type {parameter_value: overwrite_value}`)

sensor mounting: there is a boolean to autoconfigure the mounting position. if set to true (default) it will use the tfs. If not, it will use the parameters from the service request